lib/cocoapods-core/source.rb

```diff
@@ -399,7 +399,7 @@ def to_yaml
     #-------------------------------------------------------------------------#
 
     def ensure_in_repo!
-      return if Pathname.pwd.realpath == repo.realpath
+      return if Pathname.pwd.realpath.to_s.downcase == repo.realpath.to_s.downcase
```

Comment:
This should probably use `File.identical?` in case the user is actually on a case-sensitive FS